### PR TITLE
conmon: close FDs before creating exit file

### PR DESCRIPTION
**Pull request by giuseppe**
fix a race with Podman restart where it tries to bind ports
immediately after the exit file was created.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: src/conmon.c [hunk 1]
```diff
@@ -1616,23 +1617,35 @@ int main(int argc, char *argv[])
...
-			continue;
-		if (close(fd) < 0 && errno == EBADF)
-			break;
+	fdsdir = opendir("/proc/self/fd");
+	if (fdsdir != NULL) {
+		int fd;
+		int dfd = dirfd(fdsdir);
+		struct dirent *next;
+
+		for (next = readdir(fdsdir); next; next = readdir(fdsdir)) {
```
Comment (haircommander) [inline src/conmon.c]: I would prefer to declare next here instead of above. feels more idiomatic

**Comment by haircommander** [comment]
I the /proc/self/fd approach better! besides the nit LGTM

**Comment by rhatdan** [comment]
@giuseppe Make @haircommander change and then we can merge.

File: src/conmon.c (hunk 1 above)
Comment (giuseppe) [reply src/conmon.c]: I am fine with doing the change, but the current version follows more the rest of the `conmon.c` file where we don't use any C99 for loop (in facts we are using only `//` comments from C99)

Comment (haircommander) [reply src/conmon.c]: Hm okay i'd be fine with leaving it then, and instead update it all together in a future PR

**Pull request merged**